Remove the need to call modifier on exported modifier

madnificent opened the pull request:
It was previously deemed necessary to call Modifier.modifier on the exported modifier class.  We can remove this requirement based on the input of @chancancode in #7.  This commit implements the behaviour as we understood it from this issue.

The deprecation message indicates version 1.0.0 will drop support for calling Modifier.modifier, this has not been discussed and is subject to change.

The examples of the README have been updated to remove the deprecated Modifier.modifier call.

The tests have been duplicated to contain both the calls with Modifier.modifier as well as those without.  The tests have been updated mechanically with Emacs Macro's rather than being smart about them to keep them obvious.

madnificent committed:
Remove the need to call modifier on exported modifier

It was previously necessary to call Modifier.modifier on the exported
modifier class.  We can remove this requirement based on the input of
@chancancode in #7.  This commit implements the behaviour as we
understood it from this issue.

The deprecation message indicates version 1.0.0 will drop support for
calling Modifier.modifier, this has not been discussed and is subject
to change.

The examples of the README have been updated to not show the
deprecated Modifier.modifier call.

The tests have been duplicated to contain both the calls with
Modifier.modifier as well as those without.  The tests have been
updated mechanically with Emacs Macro's rather than being smart about
them to keep the tests obvious.

sukima commented on addon/-private/modifier-classic.js:
```diff
@@ -12,8 +13,11 @@ const Modifier = EmberObject.extend({
   }
 });
 
+Ember._setModifierManager(createManager, Modifier);
```
Has this been replaced with a non-underscored method (via an import)? If so, should we make that change in this PR?

madnificent replied:
Let's separate it out so we don't block this PR.

sukima commented on tests/integration/modifier-managers/oo-modifiers-classic-test.js:
```diff
@@ -13,11 +13,55 @@ module('Integration | Modifier Manager | oo modifier (classic)', function(hooks)
+    test('it has DOM element on this.element', async function(assert) {
+      this.registerModifierClass(
+        'songbird',
+        Modifier.modifier(
```
Are we supposed to still use that method here?

madnificent replied:
There's duplication in the tests now.  We call those with Modifier.modifier too, to ensure nothing breaks.

sukima commented on tests/integration/modifier-managers/oo-modifiers-classic-test.js:
```diff
@@ -193,51 +393,107 @@ module('Integration | Modifier Manager | oo modifier (classic)', function(hooks)
+          didInsertElement() {
+            assert.equal(this.testService.value, 'test-service-value');
+          }
+        })
```
Should we be testing that `Modifier.modifier` has been deprecated?

madnificent replied:
I've asked how to do this in Ember Discord#topic-testing.  If there's a reasonable response on how to test this, I'll add that test, otherwise I'll press that merge button.

sukima commented on tests/integration/modifier-managers/oo-modifiers-native-test.js:
```diff
@@ -194,53 +382,109 @@ module('Integration | Modifier Manager | oo modifier (native)', function(hooks)
+        }
+      );
+      await render(hbs`<h1 {{songbird}}>Hello</h1>`);
+    });
```
Great job with adding so many more tests!!!

madnificent committed:
Add test for deprecation

Adding tests which verify the deprecation warning thrown when using
Modifier.modifier.  The implementation is roughly based on
https://github.com/kaliber5/ember-bootstrap/pull/783/files#diff-3ac7aaa60d8ae7248dd7f234a5c8aba1R19
as suggested by @simonihmig in Ember's Discord.

sukima commented on tests/integration/modifier-managers/oo-modifiers-classic-test.js:
```diff
@@ -4,11 +4,24 @@ import { render, settled } from '@ember/test-helpers';
   hooks.beforeEach(function() {
+    deprecations = [];
+    if (!isRegistered) {
+      registerDeprecationHandler((message, options, next) => {
```
Wow, Ember gives you a register function but not an unregister function?!

Since this is essentially a global state in a test how do you feel about moving the registration to a test helper? Maybe something like this:

```js
setupDeprecartionTest(hooks);
```

Maybe look like:

```js
// tests/helpers/deprecations.js
import { registerDeprecationHandler } from '@ember/debug';

let deprecations;

registerDeprecationHandler((message, options, next) => {
  if (deprecations) {
    deprecations.push(message);
  } else {
    next(message, options);
  }
});

export default function(hooks) {
  hooks.beforeEach(function() {
    deprecations = [];
    this.deprecations = deprecations;
  });

  hooks.afterEach(function() {
    deprecations = null;
  });
}
```
